knowledge_graph_class.py

```diff
@@ -0,0 +1,94 @@
+"""Knowledge Graph"""
+from typing import Dict, List, Any
+from transformers import (
+    Text2TextGenerationPipeline,
+    AutoModelForSeq2SeqLM,
+    AutoTokenizer)
+from transformers.pipelines.text2text_generation import ReturnType
+from pydantic import Field
+
+class KnowledgeGraph():
+  """Knowledge Graph class"""
+
+  max_new_tokens: int = Field(
+    default=200, description="Maximum number of new tokens to generate."
+  )
+  predictions : List[Dict[str, Any]] = Field(default=[])
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `KnowledgeGraph` class has two class variables `max_new_tokens` and `predictions`. It's not clear why these are class variables instead of instance variables. If they are intended to be shared across all instances of the class, this could lead to unexpected behavior if multiple instances are used concurrently. If they are meant to be specific to each instance, they should be moved to the `__init__` method.

```diff
-  max_new_tokens: int = Field(
-    default=200, description="Maximum number of new tokens to generate."
-  )
-  predictions : List[Dict[str, Any]] = Field(default=[])

+  def __init__(
+    self,
+    *args,
+    model_name:str='Babelscape/rebel-large',
+    **kwargs):
+
+    self.max_new_tokens = 200
+    self.predictions = []
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->